test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,35 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var inheritance = mocks.mockInheritance();
+
+            // Act
+            objectMaker.functions = {};
+            objectMaker.processFunctions(inheritance, Object, "Object");
+            var actualMapping = objectMaker.functions;
```

Comment:
Nit: if you only use this variable once, and it's just a member of an object, it's a little cleaner IMO to put it directly in the `expect` call.